src/preferences/hotcuecolorpalettesettings.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "preferences/usersettings.h"
+#include "util/color/colorpalette.h"
+
+class HotcueColorPaletteSettings {
+  public:
+
+    enum class PaletteIndex {
+        SkinDefault = -3,
+        ByHotCueNumber = -2,
+        OffPalette = -1
+    };
+
+    explicit HotcueColorPaletteSettings(const UserSettingsPointer& pConfig)
+            : m_pConfig(pConfig) {
+    }
+
+    ColorPalette getHotcueColorPalette() const;
+
+    void setHotcueColorPalette(const ColorPalette& colorPalette);
+
+    QRgb getDefaultColor(int hotcue, bool* keepDefault = nullptr) const;
+
+    QRgb offPaletteDefaultColor() const;
+
+  private:
+    static const QString sGroup;
```

Comment:
The static const member is only needed for the inlined function. I suggest to move it into an anonymous namespace in the .cpp file and un-inline the member function. This is a typical example of premature optimization.